### PR TITLE
Revert back to old networking layer

### PR DESCRIPTION
**Pull request by aitorvs**
Task/Issue URL: https://app.asana.com/0/488551667048375/1203039959143641

### Description
TL;DR revert back to use old networking layer. Some code from #2273 is reverted.

Some new code that was added is disabled for now, eg. new networking layer stack or code from #2307

Full context in [this asana task](https://app.asana.com/0/72649045549333/1203039959143647).

### Steps to test this PR
_Smoke tests_
- [x] install from this branch
- [x] launch and enable AppTP
- [x] open apps with trackers and ensure it blocks properly

_Test App update_
- [x] install from develop,launch and enable AppTP
- [x] ensure the selected networking layer is the "new" one, ie.
 - [x] filtering logcat with `TcpDeviceToNetwork` should not show any logs when opening apps which traffic goes through VPN
 - [x] filtering logcat with `NetGuard` should show logs when opening apps which traffic goes through VPN
- [x] upgrade from this branch
- [x] launch and re-enable AppTP if necessary
- [x] filtering logcat with `TcpDeviceToNetwork` should show any logs when opening apps which traffic goes through VPN

_Test Private DNS_
- [x] install from this branch
- [x] launch and enable AppTP
- [x] open some apps to show some tracking attempts
- [x] go to Android settings -> Network & Internet -> Private DNS
- [x] In the dialog, select "Private DNS provider hostname" and input "one.one.one.one"
- [x] verify AppTP restarts
- [x] open some apps to show some tracking attempts
- [x] configure Private DNS back to "off"
- [x] verify AppTP restarts
- [x] open some apps to show some tracking attempts

_Test networkg switches_
- [x] install from this branch
- [x] launch and enable AppTP
- [x] open some apps to see tracking attempts
- [x] switch from WIFI to cellular or celluar to WIFI
- [x] Verify AppTP does NOT restart
- [x] open some apps to see tracking attempts

### CODE REVIEW

**Comment by aitorvs** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #2321** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/2321" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/2321?utm_source=stack-comment).

File: app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/integration/NgVpnNetworkStack.kt [hunk 1]
```diff
@@ -36,34 +33,21 @@ import com.duckduckgo.mobile.android.vpn.store.VpnDatabase
...
-@ContributesBinding(
-    scope = VpnScope::class,
-    boundType = VpnNetworkCallback::class
-)
-@SingleInstanceIn(VpnScope::class)
-class NgVpnNetworkStack @Inject constructor(
+
+// Make this class private to make sure we don't use it for now
+// we'll use it once we start testing the new VPN networking layer
+private class NgVpnNetworkStack constructor(
```
Comment (anikiki) [inline app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/integration/NgVpnNetworkStack.kt:43]: :+1:
